apps/api/src/server/applications/application/application.controller.js

```diff
@@ -49,17 +50,49 @@ export const createApplication = async (request, response) => {
 export const updateApplication = async ({ params, body }, response) => {
 	const mappedApplicationDetails = mapCreateApplicationRequestToRepository(body);
 
-	const updateResponse = await caseRepository.updateApplication({
+	const originalResponse = await caseRepository.getById(params.id, {
+		subSector: true,
+		sector: true,
+		applicationDetails: true,
+		zoomLevel: true,
+		regions: true,
+		caseStatus: true,
+		serviceCustomer: true,
+		serviceCustomerAddress: true,
+		gridReference: true
+	});
+
+	if (!originalResponse) {
+		throw new BackOfficeAppError(`Application not found with id ${params.id}`, 404);
+	}
+
+	let updateResponse = await caseRepository.updateApplication({
 		caseId: params.id,
 		applicantId: head(body?.applicants)?.id,
 		...mappedApplicationDetails
 	});
 
 	if (!updateResponse) {
-		throw new Error('Application not found');
+		throw new BackOfficeAppError('Application not found', 500);
+	}
+
+	const publishableFieldsChanged = publishedFieldsHaveChanged(originalResponse, updateResponse);
```

Comment:
Is it worth checking `hasUnpublishedChanges`? If it's already true we don't need to check again.